### DISCUSSION TITLE
Day view Event and Navigation View

### DISCUSSION THREAD

**Discussion by vt-ombalakumar** [ME-MarvinE/XCalendar#152] [Q&A]
Hi,
I like to create a day view events with navigation view in maui
When user click right button in navigation view, it must go to next day and show that day events below
When user click left button in navigation view, it must go to previous day and show that day event below
Can we achieve this using xcalendar?
Thanks in advance

**Comment by ME-MarvinE** [comment]
Hello,

I'm struggling to visualise what you're describing. Could you provide a simple picture with what you want this to look like?
Are you aiming for something like the "Event Calendar" example in the [sample app examples](https://github.com/ME-MarvinE/XCalendar/wiki/Sample-App)?

Or do you mean something like


| <   **1st December 2023**   > |
| ---------|
| **4PM** - Cycling with friends |
| **7PM** - Pub with friends |
| **8:45PM** - Haircut |

and after you press the right arrow you get:

| <   **2nd December 2023**   > |
| ---------|
| **3:15PM** - Doctor's appointment |
| **7PM** - Ice skating |

**Comment by vt-ombalakumar** [comment]
Yes, the same scenario you have mentioned above

**Comment by ME-MarvinE** [comment]
Definitely achievable with XCalendar.

So all you need is a NavigationView with `Text` bound to a property like `ActiveDateTime` and a CollectionView beneath it with `ItemsSource` bound to a property like `DayEvents`.

 Set the `ForwardsArrowCommand` and `BackwardsArrowCommand` properties of the NavigationView to a command that will increase or decrease the `ActiveDateTime` by one day. Then, query your list of events or your database for the events that occur on that date and update the `DayEvents` property with the results.

**Reply by vt-ombalakumar** [reply]
Let me try that, thanks

Is there any options to show calendar with day view, weekly view and monthly view in Xcalendar

**Reply by ME-MarvinE** [reply]
The Calendar is made up of DayView elements so you can use one of those.

You can change the CalendarView to show weeks by setting the Calendar's `AutoRows` property to `false` and then changing the `Rows` property to how many weeks you want. For example for 2 weeks, you would set it to `2`. You would also set the `PageStartMode` to `FirstDayOfWeek`.

If you want to show all the days in the month, that's the default functionality, and it happens when `AutoRows` is set to true. Some months have 4 weeks, some 5 weeks and some 6 weeks, but if you want to not have this variation, you can set `AutoRowsIsConsistent` to `true`.

If you mean MonthView as in showing "January", " February", "March", no, there is no option for that. But I have considered it as a potential feature along with year and decade view.

There is more info on the [Displaying Dates](https://github.com/ME-MarvinE/XCalendar/wiki/Displaying-Dates) page on the [wiki](https://github.com/ME-MarvinE/XCalendar/wiki) which could also have answers to other questions you may have.
